### PR TITLE
Drive all configuration through yaml documents

### PR DESCRIPTION
**Pull request by mark-burnett**
* Unifies join & genesis scripts as `up.sh`
* Generates `up.sh` based on configuration (e.g. for proxy or docker version)
* Makes promenade image configurable
* Puts the generated `up.sh` into a configmap for Drydock consumption

### CODE REVIEW

**Comment by mark-burnett** [comment]
Rebased onto recent documentation updates.

**Review by sh8121att — COMMENTED** [review]
LGTM, will work to provide Drydock-supporting changes so we can merge concurrently.

File: docs/getting-started.md [hunk 1]
```diff
@@ -23,52 +23,52 @@ vagrant up
...
+vagrant ssh n3 -c 'sudo /vagrant/configs/up.sh /vagrant/configs/n3.yaml'
 ```
 
 ### Development Cleanup
 
-If you are testing/developing on hosts that cannot be easily destroyed, you may
-find the `cleanup.sh` script useful.
+To use Promenade from behind a proxy, simply add proxy settings to the
+promenade `Network` configuration document using the keys `http_proxy`,
+`https_proxy`, and `no_proxy` before running `generate`.
```
Comment (alanmeadows) [inline docs/getting-started.md]: Should we provide examples here? It's important no_proxy contain things like `kubernetes` and so on that may not be immediately obvious to end users.

Comment (mark-burnett) [reply docs/getting-started.md]: I added a short statement about this in the docs.

**Pull request merged**